docs/conceptual/core/consensus.md

```diff
@@ -2,14 +2,104 @@
+
+Nodes that are following the rules of the consensus protocol always produce new blocks on top of the chain that the fork choice rule tells them to. But blockchains still must tolerate nodes that cheat and build blocks in the wrong place. Considering the example fork choice rules we studied above, it's easy to see that an attacker who wanted to revert many blocks worth of transactions could do so by building a longer chain.
+
+To prevent malicious nodes from creating attack chains and overwhelming the honest chain whenever they please, block production must be throttled so that nodes can only produce blocks at a given rate. There are two common ways of achieving this throttling.
+
+### Proof of Work
+In proof of work systems like Bitcoin, any node may produce a block at any time, so long as they have solved a computationally-intensive problem. Solving the problem takes cpu time, and thus miners can only produce blocks in proportion with their computing resources. Substrate provides a Proof of Work consensus engine.
+
+### Slots
+When using a slot based consensus algorithm, there must be a known set of validators who are permitted to produce blocks. Time is divided up into discrete slots, and during each slot only some of the validators may produce a block. The specifics of which validators can author blocks during each slot vary from engine to engine. Substrate provides Aura and Babe, both of which are slot-based block authoring engines.
+
+## Finality
+Transactors in any system want to know when their transaction is finalized, and blockchain is no different. In some traditional systems, finality happens when a receipt is handed over, or papers are signed.
+
+Using the block authoring schemes and fork choice rules described so far, transactions are never truly finalized with certainty. There is always a chance that a longer (or heavier) chain will come along and revert your transaction. However, the more blocks are built on top of a particular block, the less likely it is to ever be reverted. In this way block authoring along with a proper fork choice rule provide probabilistic finality.
+
+When true finality is desired, an additional game, known as a finality gadget can be added to the blockchain's logic. In such a game, members of a fixed authority set cast finality votes, and when a certain threshold is reached, the block is deemed finalized, and will never be reverted.
+
+> There are systems that couple block authoring more tightly with finality by considering the act of building a block as a finality vote for that block. However the two can always be separated.
+
+In systems that use a finality gadget, the fork choice rule must be modified to consider the results of the finality game. For example, instead of taking the longest chain period, a node would take the longest chain that contains the most recently finalized block. Substrate provides the Grandpa finality gadget.
```

Comment:
```suggestion
In systems that use a finality gadget, the fork choice rule must be modified to consider the results of the finality game. For example, instead of taking the longest chain period, a node would take the longest chain that contains the most recently finalized block. Substrate provides the GRANDPA finality gadget.
```